File: python/2017/1/solution2.py
with open('../../../data/2017/1/data.txt', 'rb') as f:
    captcha = f.read().strip()
captcha += captcha

total = 0
n = len(captcha) / 4
for i in xrange(2*n):
    if captcha[i] == captcha[i+n]:
            total += int(captcha[i])

print "Solution is %d" % total

File: julia/2017/1/solution2.jl
f = open("../../../data/2017/1/data.txt")
captcha = readstring(f)
close(f)
captcha = replace(captcha ^ 2, "\n", "")

total = 0
n = Int(length(captcha) / 4)
for i = 1:(2*n-1)
    if captcha[i] == captcha[i+n]
        total += parse(Int, captcha[i])
    end
end

println("Solution is $total")
